akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -142,14 +142,27 @@ private[akka] final case class Take[T](count: Long) extends PushPullStage[T, T]
 /**
  * INTERNAL API
  */
-private[akka] final case class Drop[T](count: Long) extends PushStage[T, T] {
-  private var left: Long = count
+private[akka] final case class Drop[T](count: Long) extends SimpleLinearGraphStage[T] {
+  override def initialAttributes: Attributes = DefaultAttributes.drop
 
-  override def onPush(elem: T, ctx: Context[T]): SyncDirective =
-    if (left > 0) {
-      left -= 1
-      ctx.pull()
-    } else ctx.push(elem)
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) {
+    private var left: Long = count
+
+    setHandler(in, new InHandler {
+      override def onPush(): Unit = {
+        if (left > 0) {
+          left -= 1
+          pull(in)
+        } else push(out, grab(in))
```

Comment:
You can do `passAlong(in, out)` which would set the two ports in pass-through mode, removing the branch.